src/include/daos_srv/security.h

```diff
@@ -37,9 +37,28 @@ ds_sec_alloc_default_daos_pool_acl(void);
  *
  * \return	Newly allocated struct daos_acl
  */
+
 struct daos_acl *
 ds_sec_alloc_default_daos_cont_acl(void);
 
+/**
+ * Obtain the origin name specified by the security credential.
+ *
+ * \param[in]	cred		User's security credential
+ * \param[out]  machine		Hostname of the machine that generated the credential.
+ *
+ * \return	0		Success
+ *		-DER_INVAL	Invalid input
+ *		-DER_BADPATH	Can't connect to the control plane socket at
+ *				the expected path
+ *		-DER_NOMEM	Out of memory
+ *		-DER_NOREPLY	No response from control plane
+ *		-DER_MISC	Error in control plane communications
+ *		-DER_PROTO	Unexpected or corrupt payload from control plane
+ */
+int
+ds_sec_pool_get_origin(d_iov_t *cred, char **machine);
```

Comment:
How about `ds_sec_cred_get_machine`?

Reply:
Origin matches the name in the credential protobuf message. In the auth_sys token it happens to be a machine name. It could conceivably be something else. Origin sounded weird in the command line even though its what we call it here.

Reply:
Okay, I can buy that, although the comments seem to indicate origin and machine are interchangeable, rather than machine being a subset of origin types. The name still should be "cred_get_origin" instead of "pool_get_origin" though.

Reply:
Ok renamed